spec/services/service_offering_spec.rb

```diff
@@ -0,0 +1,44 @@
+describe ServiceOffering do
+  let(:api_instance) { double(:api_instance, :show_service_offering => service_offering) }
+  let(:service_offering) { ServiceOffering.new(params) }
+  let(:params) { {'service_offering_ref' => '10'} }
+
+  it "#{described_class}#find" do
+    with_modified_env TOPOLOGY_SERVICE_URL: 'http://www.example.com' do
+      allow(service_offering).to receive(:api_instance).and_return(api_instance)
```

Comment:
Ok so I think the only reason we need the `with_modified_env` is for the initialization of the `service_offering`, right?

In that case, I'm pretty sure we can move the `with_modified_env` block along with the `allow` stub into a before block, something like this?

```ruby
describe "#self.find" do
  before do
    with_modified_env TOPOLOGY_SERVICE_URL: 'http://www.example.com' do
      allow(service_offering).to receive(:api_instance).and_return(api_instance)
    end
  end

  it "creates a new service_offering and returns its id" do
    expect(described_class).to receive(:new).with({}).and_return(service_offering)
    ServiceOffering.find(1)
  end
end
```

Reply:
Possibly the above code works, I tried a `before do` yesterday and ran into issues.  I'll take a look again.

Reply:
This is resolved.  I attempted to take it one step farther and add `with_modified_env` into its own helper file as well, so I could eliminate the block in the `before` block, but ran into issues, I believe it had to do with the object being passed into the `with_modified_env` wasn't referenced as I thought it would be.

Also, on the helper methods. The infrastructure, and other helpers existed in https://github.com/ManageIQ/service_portal-api/pull/12 so was waiting for that to be merged. Either way, I built new helpers in here and moved my and Madhu's duplicated callers into that helper.